webapp/src/ts/modules/analytics/analytics-route-guard.provider.ts

```diff
@@ -0,0 +1,31 @@
+import { ActivatedRouteSnapshot, Router, CanActivateChild, UrlTree } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+import { Selectors } from '@mm-selectors/index';
+
+@Injectable()
+export class AnalyticsRouteGuardProvider implements CanActivateChild {
+
+  constructor(
+    private store: Store,
+    private router: Router
+  ) { }
+
+  canActivateChild(route: ActivatedRouteSnapshot): Observable<boolean | UrlTree> {
+    return this.store
+      .select(Selectors.getAnalyticsModules)
+      .pipe(
+        filter(modules => !!modules),
```

Comment:
If we update the reducer to initialize and always return a valid value, we don't need this filter.

Reply:
I wanted to take the emit from when we set values from the service (2nd emit) and not the default value (1st emit, default is null).

Reply:
I think I'm going to try the suggestion of adding the fallback when setting a new value in the reducer, but the initial value stays in `null`.

```js
// Analytics reducer
const initialState = {
  selected: null,
  analyticsModules: null
};

...
...

on(Actions.setAnalyticsModules, (state, { payload: { analyticsModules } }) => {
    return { ...state, analyticsModules || [] };
})
```
`null` allows us to know when the reducer hasn't set a value yet, that we're waiting for data, then this pipe's filter will prevent resolving the route guard too soon when loading the app and analytics is the selected tab. 👍

Reply:
Done